### ISSUE TITLE
EL8: vips-devel depends on both IM6 and IM7

### ISSUE THREAD

**Issue by kleisauke** [remicollet/remirepo#134]
It looks like `vips-devel` depends on both `ImageMagick-devel` and `ImageMagick7-devel`.
```bash
$ dnf repoquery --whatrequires ImageMagick7-devel | grep vips
vips-devel-0:8.8.3-3.el8.remi.x86_64
vips-devel-0:8.8.4-1.el8.remi.x86_64
vips-full-devel-0:8.8.3-4.el8.remi.x86_64
vips-full-devel-0:8.8.4-2.el8.remi.x86_64
$ dnf repoquery --requires --resolve vips-devel | grep Magick
ImageMagick-devel-1:6.9.10.82-1.el8.remi.x86_64
ImageMagick7-devel-0:7.0.9.12-1.el8.remi.x86_64
$ dnf repoquery --requires vips-devel | grep Magick
pkgconfig(MagickCore)
$ dnf repoquery --whatprovides "pkgconfig(MagickCore)"
ImageMagick-devel-0:6.9.10.64-1.el8.x86_64
ImageMagick-devel-1:6.9.10.79-1.el8.remi.x86_64
ImageMagick-devel-1:6.9.10.82-1.el8.remi.x86_64
ImageMagick7-devel-0:7.0.9.12-1.el8.remi.x86_64
ImageMagick7-devel-0:7.0.9.9-1.el8.remi.x86_64
$ dnf repoquery --whatprovides "pkgconfig(MagickCore) < 7"
ImageMagick-devel-0:6.9.10.64-1.el8.x86_64
ImageMagick-devel-1:6.9.10.79-1.el8.remi.x86_64
ImageMagick-devel-1:6.9.10.82-1.el8.remi.x86_64
```

Is this a packaging bug? It would be nice if it only depends on the ImageMagick6 development files.

**Comment by remicollet** [comment]
Sorry for late answer, it seems I miss the notification

All the pkgconfig dependencies are auto generated  during rpmbuild....
Having to filter them will be..... strange.

Need to think about this

**Comment by kleisauke** [comment]
Don't worry, it's just a minor issue I ran into. How about enforcing IM6 within the vips spec file?

Something like this: 
```diff
diff --git a/vips.spec b/vips.spec
--- a/vips.spec
+++ b/vips.spec
@@ -47,7 +47,7 @@ BuildRequires:	pkgconfig(glib-2.0)
 BuildRequires:	pkgconfig(expat)
 BuildRequires:	pkgconfig(fftw3)
 %if 0%{?fedora} >= 30 || 0%{?rhel} >= 8
-BuildRequires:	ImageMagick-devel
+BuildRequires:	pkgconfig(MagickCore) < 7
 %else
 # Ensure we use version 6 (same as imagick ext).
 BuildRequires:	ImageMagick6-devel
```
(untested, so I don't know if that actually works)

**Comment by remicollet** [comment]
Using pkgconfig is not possible, as EL-7 have 2 providers (IM 6.7.2.7 from base repo and IM 6.9.10.89 from mine), and we need to ensure consistency between buildtime and runtime version

Fixed in upcoming 8.9.1
https://git.remirepo.net/cgit/rpms/lib/vips.git/commit/?id=b2550db1566f232fda938e25d383f3aa69fcebc1

**Comment by kleisauke** [comment]
I confirm that `vips-devel` (and `vips-full-devel`) only depends on `ImageMagick-devel` with v8.9.1, thanks!
